src/Sulu/Bundle/AudienceTargetingBundle/Entity/TargetGroup.php

```diff
@@ -72,85 +54,85 @@ public function __construct()
         $this->rules = new ArrayCollection();
     }
 
-    public function getId()
+    public function getId(): int
     {
         return $this->id;
     }
 
-    public function getTitle()
+    public function getTitle(): string
     {
         return $this->title;
     }
 
-    public function setTitle($title)
+    public function setTitle(string $title): static
     {
         $this->title = $title;
 
         return $this;
     }
 
-    /**
-     * @return string|null
-     */
-    public function getDescription()
+    public function getDescription(): ?string
     {
         return $this->description;
     }
 
-    public function setDescription($description)
+    public function setDescription(?string $description): static
     {
         $this->description = $description;
 
         return $this;
     }
 
-    public function getPriority()
+    public function getPriority(): int
     {
         return $this->priority;
     }
 
-    public function setPriority($priority)
+    public function setPriority(int $priority): static
     {
         $this->priority = $priority;
 
         return $this;
     }
 
-    public function isAllWebspaces()
+    public function isAllWebspaces(): bool
     {
         return $this->allWebspaces;
     }
 
-    public function setAllWebspaces($allWebspaces)
+    public function setAllWebspaces(bool $allWebspaces): static
     {
         $this->allWebspaces = $allWebspaces;
 
         return $this;
     }
 
-    public function isActive()
+    public function isActive(): bool
     {
         return $this->active;
     }
 
-    public function setActive($active)
+    public function setActive(bool $active): static
     {
         $this->active = $active;
 
         return $this;
     }
 
-    public function getWebspaces()
+    public function getWebspaces(): Collection
     {
         return $this->webspaces;
     }
 
+    /**
+     * @return string[]
+     */
     #[VirtualProperty]
     #[SerializedName('webspaceKeys')]
-    public function getWebspaceKeys()
+    public function getWebspaceKeys(): array
     {
-        // @phpstan-ignore-next-line
-        if (null === $this->webspaces) {
+        // @phpstan-ignore isset.initializedProperty (property may be uninitialized when loaded from trash)
+        if (!isset($this->webspaces)) {
```

Comment:
I think the comment is misleading. The only case I can think of is deserialization where this webspaces could not be defined 🤔 

It looks even that the audience targeting bundle doe snot even have a trash implementation.